core/src/main/scala/kafka/utils/KafkaScheduler.scala

```diff
@@ -88,12 +88,14 @@ class KafkaScheduler(val threads: Int,
   
   override def shutdown() {
     debug("Shutting down task scheduler.")
-    this synchronized {
-      if(isStarted) {
-        executor.shutdown()
-        executor.awaitTermination(1, TimeUnit.DAYS)
+    // We use the local variable to avoid NullPointerException if another thread shuts down scheduler at same time.
```

Comment:
@jjkoshy  Good idea, a `shutdownLock` will make it more clear. I'll do that. Thanks.